src/basic/env-file.c

```diff
@@ -414,30 +414,39 @@ static int load_env_file_push_pairs(
                 const char *key, char *value,
                 void *userdata,
                 int *n_pushed) {
-        char ***m = userdata;
+        char ***m = ASSERT_PTR(userdata);
+        bool added = false;
         int r;
 
         r = check_utf8ness_and_warn(filename, line, key, value);
         if (r < 0)
                 return r;
 
+        /* Check if the key is present */
+        for (char **t = *m; t && *t; t += 2)
+                if (streq(t[0], key)) {
+                        if (value)
+                                r = free_and_replace(t[1], value);
+                        else
+                                r = free_and_strdup(t+1, "");
+                        goto fini;
```

Comment:
what happened to `…sh`?

Reply:
in 99% of the cases we call the label we jump to for cleaning up "finish". But here you called it "fini", i.e. the "sh" somehow vanished. And I wondered where it went.

Reply:
Oh, I see. No particular reason, I can make it "finish".